File: billing-application/src/main/java/com/example/billing/application/port/in/WalletQueryUseCase.java
package com.example.billing.application.port.in;

import com.example.billing.domain.ledger.LedgerEntry;
import com.example.billing.domain.wallet.Wallet;

import java.util.List;

public interface WalletQueryUseCase {
    Wallet getByOwner(String ownerId);
    List<LedgerEntry> recentTransactions(String ownerId, int limit);
}

File: billing-application/src/main/kotlin/com/example/billing/application/port/in/WalletQueryUseCase.kt
package com.example.billing.application.port.`in`

import com.example.billing.domain.ledger.LedgerEntry
import com.example.billing.domain.wallet.Wallet

interface WalletQueryUseCase {
    fun getByOwner(ownerId: String): Wallet
    fun recentTransactions(ownerId: String, limit: Int): List<LedgerEntry>
}
